core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -436,6 +440,8 @@ private void commitSimpleTransaction() {
     } catch (RuntimeException e) {
       if (!ops.requireStrictCleanup() || e instanceof CleanableFailure) {
         cleanUpOnCommitFailure();
+      } else if (ops.requireStrictCleanup()) {
+        throw new CommitStateUnknownException(e);
```

Comment:
Maybe we can have a helper function cleanupIfPossible(e, () -> someCleanupFunction()) to reduce the duplication here

Reply:
Yeah I probably should, it's failing the cyclomatic check here